src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -159,6 +157,15 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
                 throw new ArgumentNullException(nameof(formatters));
             }
 
+            var responseContentType = formatterContext.HttpContext.Response.ContentType;
+            if (!string.IsNullOrEmpty(responseContentType))
+            {
+                contentTypes.Clear();
+                contentTypes.Add(MediaTypeHeaderValue.Parse(responseContentType));
+            }
```

Comment:
Why does `Response.ContentType` take precedence over something set on the object result? Let's get agreement on the right priority, and add some comments here to that effect. Let's discuss with @lodejard when he's around.